openmdao/jacobians/assembled_jacobian.py

```diff
@@ -357,6 +357,11 @@ def _update(self, system):
         if ext_mtx is not None:
             ext_mtx._post_update()
 
+        if self._under_complex_step:
+            # If we create a new _int_mtx while under complex step, we need to convert it to a
+            # complex data type.
+            self._int_mtx.set_complex_step_mode(True)
```

Comment:
Does the _int_mtx need to get set back to float at some point?

Reply:
It does, and that happens as normal when `_set_complex_step_mode` is called at the conclusion of the approximation.